sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListener.java

```diff
@@ -152,5 +162,36 @@ private int onConnectionEvent(final RntbdConnectionEvent event, final Throwable
 
         return 0;
     }
+
+    public void attemptBackgroundAddressRefresh(RxDocumentServiceRequest request) {
+
+        if (request.requestContext == null) {
+            return;
+        }
+
+        if (request.requestContext.isRequestCancelledOnTimeout() == null) {
+            return;
+        }
+
+        AtomicBoolean isRequestCancelledOnTimeout = request.requestContext.isRequestCancelledOnTimeout();
+        final boolean forceAddressRefresh = request.requestContext.forceRefreshAddressCache;
+
+        if (this.forceBackgroundAddressRefreshInProgress.compareAndSet(false, true)
```

Comment:
if the request landed here with isRequestCancelledOnTimeout=false, then I think forceBackgroundAddressRefreshInProgress is never be reset currently

Reply:
Fixed in next iteration.